ITL-4 Implement base state of todo card

BiroAron opened the pull request.

BiroAron committed:
ITL-7 Create the header component

BiroAron committed:
ITL-8 Add the title

BiroAron committed:
ITL-9 Add an add button, include a icon library or include the icon from the design

BiroAron committed:
ITL-10 Create the event that triggers the todo creation functionality

BiroAron committed:
Replaced plain CSS with Tailwind

BiroAron committed:
Made all the changes regarding to the comments

BiroAron committed:
Made the changes regarding to class names and reactive variables

BiroAron committed:
Removed all unnecessary code and comments

BiroAron committed:
Removed unnecessary id-s and redundant script

BiroAron committed:
Merge branch 'feature/ITL-1-implement-page-header' into develop

BiroAron committed:
ITL 11 Create the Placeholder img when the user is in the home page and no todos added to todo list

BiroAron committed:
ITL 12 Mobile friendly looking

BiroAron committed:
ITL 13  make the todo list look

BiroAron committed:
Made changes regarding the comments

-removed all of the conditional operators from the template tags
-removed unnecessary complicated code
-renamed non-descriptive variable names
-created a colorMap

BiroAron committed:
updatePriorityClass is now a variable instead of a function

BiroAron committed:
Merge branch 'feature/ITL-2-implement-todo-list' into develop

BiroAron committed:
ITL14 Subtask : Create an empty to-do item

BiroAron committed:
ITL 15Subtask 2: Display the new to-do item at the top of the list

BiroAron committed:
ITL 16Subtask 3: Set proper placeholder text for the new to-do item

BiroAron committed:
Created the requested file system

BiroAron committed:
ITL 17 Subtask 4 :After tapping the task, Card will convert it into edit mode

BiroAron committed:
ITL 19  Subtask 6 : Make the design match with figma

BiroAron committed:
ITL 20 Subtask 7: Pop-up confirmation

BiroAron committed:
Cleaned up the code

BiroAron committed:
Made changes regarding the comments

- removed modification of props in child components

- component API orders corrected

BiroAron committed:
Removed all reactive usage of props in child elements

BiroAron committed:
Divided the ToDoForm.vue into five subcomponents

BiroAron committed:
Merge branch 'feature/ITL-3-implement-edit-and-add-state-for-todo-card' into develop

BiroAron committed:
ITL 21 move the completed todo item to the bottom of the todo list

when the user completes a todo ,animate a checkmark  and move the completed todo item to the bottom of the todo list.

kincsokiss commented on src/components/todo/ToDoDeleteButton.vue:
```diff
@@ -0,0 +1,16 @@
+<template>
+  <button
+    class="bg-gray-300 py-2 px-7 rounded-xl font-semibold text-black flex justify-center align-center"
```
Please create a general Button.vue component, enable to set the classes, button title and the function for click event as props. Use this component as Delete and Save buttons insetad of TodoDeleteButton and TodoSaveButton components. Those Vue components are almost the same.

BiroAron committed:
Made changes regarding the comments and fixed reactivity issues

BiroAron committed:
ITL 4 - Made changes regarding the comments

-removed components named ToDoDeleteButton and TodoSaveButton components, combined them into one named Button.vue

kincsokiss commented on src/components/todo/Button.vue:
```diff
@@ -0,0 +1,30 @@
+const emit = defineEmits(['handleSaveClick', 'changePopupState'])
+
+function handleClick() {
+  if (props.buttonName === 'Save') emit('handleSaveClick')
```
What will happen if there will be further buttons? Ex. update, delete etc. ? Will you add them to this if..else statement and to the emits as well? 
I general button has not to know about what kind of button is. It receive all of the specific details as props and emits. It has a click event which has to emit a function, but the parent component has to handle this.

gerebdavid commented on src/components/todo/ToDoDescription.vue:
```diff
@@ -0,0 +1,25 @@
+<template>
+  <textarea
+    :value="description"
+    @input="handleInput"
```
move events after attributes

gerebdavid commented on src/components/todo/ToDoDescription.vue:
```diff
@@ -0,0 +1,25 @@
+<template>
+  <textarea
+    :value="description"
```
Use v-model instead of :value and create a computed with getter setter

gerebdavid commented on src/components/todo/ToDoForm.vue:
```diff
@@ -0,0 +1,126 @@
+      ></ToDoDescription>
+      <div class="flex justify-start align-center mt-4">
+        <ToDoButton
+          :buttonstyles="'bg-primary text-white'"
```
You dont have to add : to buttonstyles prop cause the value is a string

gerebdavid commented on src/components/todo/ToDoForm.vue:
```diff
@@ -0,0 +1,126 @@
+      <div class="flex justify-start align-center mt-4">
+        <ToDoButton
+          :buttonstyles="'bg-primary text-white'"
+          :button-name="'Save'"
```
same on button-name

gerebdavid commented on src/components/todo/ToDoPriority.vue:
```diff
@@ -0,0 +1,117 @@
+      :class="getDropdownState"
+    >
+      <div class="flex flex-col items-start">
+        <li>
```
Use v-for on <li> instead of creating three li element

gerebdavid commented on src/components/todo/ToDoPriority.vue:
```diff
@@ -0,0 +1,117 @@
+    </ul>
+  </div>
+  <div class="flex justify-center space-x-2 desktop:hidden">
+    <div
```
Same here

gerebdavid commented on src/components/todo/ToDoPriority.vue:
```diff
@@ -0,0 +1,117 @@
+
+function priorityBorderChange(priority: TodoPriority) {
+  if (priority === localTodo.priority) return 'border border-black border-2'
+  else return 'border-none'
```
If you return inside if you dont have to use else

gerebdavid commented on src/components/todo/ToDoTitle.vue:
```diff
@@ -0,0 +1,25 @@
+<template>
+  <input
+    :value="title"
```
Do the same like in TodoDescription

BiroAron committed:
ITL 4 - Made changes regarding the comments

- replaced Button.vue with BaseButton.vue
- replaced :value with v-model in TodoDescription and TodoTitle components
- used v-for instead of creating multiple li elements in TodoPriority component

BiroAron committed:
ITL 4 - Replaced all  basic button elements with BaseButton.vue
